src/Blazored.Toast.TestExtensions/InMemoryToastService.cs

```diff
@@ -0,0 +1,118 @@
+            => Toasts.Add(new InMemoryToast(level, message, GetHeading(level, heading)));
+
+        public void ShowWarning(string message, string heading = "", Action onClick = null)
+            => ShowToast(ToastLevel.Warning, message, heading, onClick);
+
+        public void ShowWarning(RenderFragment message, string heading = "", Action onClick = null)
+            => ShowToast(ToastLevel.Warning, message, heading, onClick);
+
+        private string GetHeading(ToastLevel level, string heading)
+        {
+            if (!string.IsNullOrWhiteSpace(heading)) return heading;
+
+            return level switch
+            {
+                ToastLevel.Error => "Error",
+                ToastLevel.Info => "Info",
+                ToastLevel.Success => "Success",
+                ToastLevel.Warning => "Warning",
+                _ => throw new InvalidOperationException(),
+            };
+        }
+        /// <summary>
+        /// Checks a specific number of toasts are displayed.
+        /// </summary>
+        /// <param name="count">The number of toast elements expected.</param>
+        public bool ToastCountIs(int count)
+        {
+            if (count < 1)
+                throw new ArgumentOutOfRangeException(nameof(count), $"{nameof(count)} value must be greater than 0.");
+
+            return Toasts.Count == count;
+        }
+
+        /// <summary>
+        /// Checks a single toast is displayed.
+        /// </summary>
+        public bool ToastCountIsOne()
+            => Toasts.Count == 1;
+
+        /// <summary>
+        /// Checks a single toast is displayed with level.
+        /// </summary>
+        /// <param name="level">The toast level expected.</param>
+        public bool ToastCountIsOneWithLevel(ToastLevel level)
+        {
+            if (Toasts.Count != 1)
+                throw new ToastCountException(1, Toasts.Count);
+
+            return Toasts.Single().HasLevel(level);
+        }
+
+        /// <summary>
+        /// Checks a single toast is displayed with heading.
+        /// </summary>
+        /// <param name="heading">The toast heading expected.</param>
+        public bool ToastCountIsOneWithHeading(string heading)
+        {
+            if (Toasts.Count != 1)
+                throw new ToastCountException(1, Toasts.Count);
+
+            return Toasts.Single().HasHeading(heading);
+        }
+
+        /// <summary>
+        /// Checks a single toast is displayed with message.
+        /// </summary>
+        /// <param name="message">The toast message expected.</param>
+        public bool ToastCountIsOneWithMessage(string message)
+        {
+            if (Toasts.Count != 1)
+                throw new ToastCountException(1, Toasts.Count);
+
+            return Toasts.Single().HasMessage(message);
+        }
+
+        /// <summary>
+        /// Checks no toast elements are displayed.
+        /// </summary>
+        public bool ToastCountIsZero()
+            => Toasts.Count == 0;
+    }
```

Comment:
I understand why want to make it easy for users to perform assertions. However, my experience is that you should avoid this, since different users will use different assertion libraries. Some use the built-in ones in e.g. xUnit and NUnit, and some like things like FluentAssertions or Shoudly.  Opinionated method names like this will work great with some assertion libraries, not all others.

My instinct would be to simply expose the `Toasts` list.

Reply:
Thanks for the feedback, I will go ahead and make the changes you suggested